### ISSUE TITLE
Solr_Configure silently ingoring server errors

### ISSUE THREAD

**Issue by mateusz** [silverstripe/cwp-search#14]
The errors are silently ignored and never surfaced to the user in [CwpSolrConfigStore::uploadString](https://github.com/silverstripe/cwp-search/blob/master/src/Solr/CwpSolrConfigStore.php#L62).

This means the user of this module will think everything is fine, while it might not be, as the infrastructure (CWP) has no way to communicate upload errors.

**Comment by robbieaverill** [comment]
@mateusz so we should add some logging for when file_get_contents fails to read in that example, sound OK?

**Comment by mateusz** [comment]
Yes, ideally you'd surface the error code (such as 400) and also the error message from the body of the response. And you might want to handle the breakage too? Does it even make sense to continue if one file failed to upload?

**Comment by robbieaverill** [comment]
We could throw an exception but that would probably be a breaking change since it'd currently fail silently. You probably know more about Solr configuration than I do, so I'm happy to follow your suggestions @mateusz =)

**Comment by mateusz** [comment]
I'm afraid I don't know as much as you'd like to believe :-) there is a lot of fragmentation in our knowledge. I don't really mind how and when it's fixed. Developers probably don't expect a silent success on failure, so it sounds a bit like a bug.

**Comment by robbieaverill** [comment]
Have had a quick look at this. Without breaking any public API contracts, I think we'd need to provide a logger to the `SolrConfigStore` implementations so they can then use them to provide logging context in a similar way to how the `Solr_Configure` task loads its logger:

```php
        $logger = $this
            ->getLoggerFactory()
            ->getOutputLogger($name, $verbose);
```

This would be a minor change in my head, so I'll re-label as such.
